### PR TITLE
chore(router-dom): upgrade react-router-dom 

### LINKED ISSUE #675: upgrade react-router-dom
# Overview

We currently use `react-router-dom` on version `5.2.0` and want to update to the latest major version, as of writing, `6.2.1`.

There is an [example closed PR here](https://github.com/shapeshift/web/pull/469) showing what is required.

This will mainly involve fixing type errors and breaking changes throughout the app and will take some time to get all of the errors fixed after the upgrade.

# Acceptance Criteria

- App is fully functional with no regressions
- CI passes, i.e. lint/type-check/build/test

# Bounty Hunters
- Join our discord [here](https://discord.gg/shapeshift).  Feel free to ask for help in #engineering-public channel
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info

### PR DESCRIPTION
**Pull request by developerfred**
## Description

Upgrade React Router 

close #675

## Ref
- https://typescript.tv/react/upgrade-to-react-router-v6/
- https://reactrouter.com/docs/en/v6/upgrading/v5#upgrade-to-react-router-v6

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn` to install new deps
2. etc...

### CODE REVIEW

**Comment by 0xdef1cafe** [comment]
@developerfred thanks so much for this - i know this is only in draft, but you can run `yarn lint:fix` to autofix lint errors.

let us know if you have any feedback/need assistance testing this - we have an ops team that can run a regression test before merging.

@Neverwas-dev this one might be coming your way for a general regression test

**Comment by developerfred** [comment]
> @developerfred thanks so much for this - i know this is only in draft, but you can run `yarn lint:fix` to autofix lint errors.
> 
> let us know if you have any feedback/need assistance testing this - we have an ops team that can run a regression test before merging.
> 
> @Neverwas-dev this one might be coming your way for a general regression test

I thank you for the opportunity to contribute, for now I'm still running and fixing the lint errors.

if necessary I will notify you.

But thinking aloud I think it would be nice to test Dapp and as a contract interaction function.

**Comment by developerfred** [comment]
done @0xdef1cafe

**Comment by 0xdef1cafe** [comment]
hey @developerfred - looking good, but unfortunately this is failing type checking in CI

are you able to run `yarn type-check` locally and fix the type errors locally and push the fixes?

cheers.

**Comment by developerfred** [comment]
@0xdef1cafe I'll take a look at that now.

**Comment by 0xdef1cafe** [comment]
hey @developerfred - just a heads up, we have a few more pages/routes coming within the next week, so we're going to hold off on merging this until those are done and in production.

we'd still like to get this upgrade done, and we're happy to adjust the bounty for the additional workload involved, but don't feel any need to rush this in within the next few days. 

we'll ping you when the additional work is completed and would love to see this get finished then if you're cool with that?

**Comment by developerfred** [comment]
> hey @developerfred - just a heads up, we have a few more pages/routes coming within the next week, so we're going to hold off on merging this until those are done and in production.
> 
> we'd still like to get this upgrade done, and we're happy to adjust the bounty for the additional workload involved, but don't feel any need to rush this in within the next few days.
> 
> we'll ping you when the additional work is completed and would love to see this get finished then if you're cool with that?

I agree, yes, I'm waiting, I think now I only had 10 warnings :D

I hope thank you and gladly accept the tip.

**Comment by 0xdef1cafe** [comment]
hey @developerfred - develop has stabilized and this one is ready to go again. you're going to have a few conflicts but we can compensate if there's significant additional work, please reach out to @0xean if that's the case 🤘

**Comment by cjthompson** [comment]
@developerfred - we're ready to have you update this PR against current develop.

**Comment by developerfred** [comment]
> we're ready to have you update this PR against current develop.

Ok, I'll update

**Comment by 0xean** [comment]
@developerfred - please ping us when you are ready for another pass on this one. Thanks again for being flexible with our team!

**Comment by developerfred** [comment]
@0xean Ok, I can get back to that at the beginning of the week, do you have a preference on another task?

**Comment by 0xean** [comment]
> 

prefernce for another issue for you to pick up? or I am not sure I am understanding the question.

**Comment by 0xean** [comment]
ahh, nope! Would love to get this PR merged once it's ready to go.  Sorry for the confusion, whenever you are able to get to it is great.

**Comment by developerfred** [comment]
> ahh, nope! Would love to get this PR merged once it's ready to go. Sorry for the confusion, whenever you are able to get to it is great.

Perfect, thanks for understanding.

File: src/Routes/Routes.tsx
Comment (0xdef1cafe) [inline src/Routes/Routes.tsx:1]: merge conflict marker

File: src/components/Button/button.stories.tsx
Comment (0xdef1cafe) [inline src/components/Button/button.stories.tsx:110]: these seem unrelated to the router changes?

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

**Comment by 0xdef1cafe** [comment]
@developerfred can you please 1. fix the merge conflict marker, and 2. revert all of the unrelated UI layer prop changes if they're not required, or use single quotes please

File: src/components/Button/button.stories.tsx
Comment (developerfred) [reply src/components/Button/button.stories.tsx:110]: @0xdef1cafe I believe "sm" to "small" and "lg" to "large"  was a setting of my global typescript.

File: src/components/SelectAssets/SelectAssetView.tsx
Comment (0xean) [inline src/components/SelectAssets/SelectAssetView.tsx:3]: @developerfred - can you take a look at this file, its failing to build in our CI.  I just pulled it down locally to also try and I am getting the following
```
SyntaxError: shapeshift/web/src/components/SelectAssets/SelectAssetView.tsx: 'return' outside of function. (41:2)
  39 |   }, [navigate])}
  40 |   
> 41 |   return (
     |   ^
  42 |   <Routes location={location} key={location.key}>
  43 |     <Route
  44 |       path={SelectAssetRoutes.Search}
```

Comment (developerfred) [reply src/components/SelectAssets/SelectAssetView.tsx]: fixed here 8affb97

Comment (0xean) [reply src/components/SelectAssets/SelectAssetView.tsx]: TY!

**Comment by 0xean** [comment]
okay, building looks good now, but its failing the `yarn type-check` will also confirm locally.

**Comment by 0xdef1cafe** [comment]
@developerfred you might want to try `yarn local-ci` to essentially the same steps CI does

**Comment by developerfred** [comment]
done @0xdef1cafe and @0xean

**Comment by 0xdef1cafe** [comment]
> done @0xdef1cafe and @0xean

getting closer! few failing tests in CI and we'll be good to get this regression tested

**Comment by developerfred** [comment]
> > done @0xdef1cafe and @0xean
> 
> getting closer! few failing tests in CI and we'll be good to get this regression tested

Test has been fixed on [d7dc0ef](https://github.com/shapeshift/web/pull/695/commits/d7dc0ef13620f04ceca3b43dbe33babc244a2279)

**Comment by 0xdef1cafe** [comment]
hey @developerfred - given we keep falling out of sync, and this needs a full regression test, we're going to close this PR, reopen it under the `shapeshift` repo and manage getting it merged internally, and pay you out for the work done on the bounty.

**Comment by developerfred** [comment]
@0xdef1cafe  ok! Thanks!

**Pull request closed without merging**
